### PR TITLE
separate links and entities in relationships

### PR DESCRIPTION
**Pull request by marius-patrascu**

### CODE REVIEW

**Comment by sarbull** [comment]
@marius-patrascu this is not really how we discussed, i mentioned you to add types/group in the odpi/egeria-js-commons and dinamically map them in the implementation under happi-graph

this way you will not manually hardcode the titles for links and nodes

File: happi-graph-legend.js [hunk 1]
```diff
@@ -190,6 +207,7 @@ class HappiGraphLegend extends PolymerElement {
       </paper-button>
 
       <template is="dom-if" if="[[ isMinimized ]]" restamp="true">
+        <div class="icon-title">Entities</div>
```
Comment (sarbull) [inline happi-graph-legend.js:210]: this should come from a groupped map by type/group

File: happi-graph-legend.js [hunk 2]
```diff
@@ -198,6 +216,18 @@ class HappiGraphLegend extends PolymerElement {
               <span>[[ item ]]</span>
             </div>
           </template>
+         
+        </div>
+        <div class="icon-title">Relationships</div>
```
Comment (sarbull) [inline happi-graph-legend.js:221]: this should come from a groupped map by type/group

File: happi-graph-legend.js [hunk 3]
```diff
@@ -27,6 +27,10 @@ class HappiGraphLegend extends PolymerElement {
         type: Object,
         value: []
       },
+      linkLabels: {
```
Comment (sarbull) [inline happi-graph-legend.js:30]: this is redundant

**Review by sarbull — CHANGES_REQUESTED** [review]

**Comment by marius-patrascu** [comment]
I updated the PR but I still have to improve the code to generate the correct title, at the moment it does not pluralize the group retrieved from egeria js commons

File: happi-graph.js [hunk 4]
```diff
@@ -668,6 +668,7 @@ class HappiGraph extends PolymerElement {
                                 graph-links="{{ links }}"
                                 icons-map="{{ iconsMap }}"
                                 properties-map="{{ propertiesMap }}"
+                                legend-data = "{}"
```
Comment (sarbull) [inline happi-graph.js:671]: this is internal property, you don't need to add it as an input

File: package.json [hunk 5]
```diff
@@ -10,17 +10,18 @@
     "test": "polymer test"
   },
   "dependencies": {
-    "@polymer/paper-button": "3.0.1",
-    "@polymer/paper-icon-button": "3.0.2",
+    "@capaj/pluralize": "git+https://github.com/capaj/pluralize.git",
     "@polymer/iron-icons": "3.0.1",
     "@polymer/iron-iconset-svg": "3.0.1",
+    "@polymer/paper-button": "3.0.1",
+    "@polymer/paper-icon-button": "3.0.2",
```
Comment (sarbull) [inline package.json:17]: these are unused dependencies, right? we can remove them

**Review by sarbull — APPROVED** [review]

**Pull request merged**